src/indexset.jl

```diff
@@ -1192,8 +1143,8 @@ NDTensors.getindices(is::IndexSet, I...) = getindices(is, I...)
 
 Return a new IndexSet with indices `setdir(is[i], dirs[i])`.
 """
-function setdirs(is::IndexSet{N}, dirs) where {N}
-  return IndexSet(ntuple(i -> setdir(is[i], dirs[i]), Val(N)))
+function setdirs(is::IndexSet, dirs)
+  return IndexSet(ntuple(i -> setdir(is[i], dirs[i]), Val(length(is))))
```

Comment:
Probably just want to replace this with a call to `map` (or maybe broadcast or comprehension, whichever is fastest), once we change to a `Vector{IndexT}` storage.